File: Cpp/1688_Count_of_Matches_in_Tournament.cpp
class Solution {
public:
    int numberOfMatches(int n) {
        int result = 0;
        int team = n;
        while(team != 1) {
            result += team/2;
            if(team % 2 == 0) team /= 2;
            else team = team/2 + 1;
        }
        
        return result;
    }
};

File: Kotlin/1688_Count_of_Matches_in_Tournament.kt
class Solution {
    fun numberOfMatches(n: Int): Int {
        var result = 0
        var team = n
        
        while(team != 1) {
            result += team / 2
            if(team % 2 == 0) team /= 2
            else team = team / 2 + 1
        }
        
        return result
    }
}